### DISCUSSION TITLE
Velocity direction reversed from biological direction

### DISCUSSION THREAD

**Discussion by skannan4** [theislab/scvelo#462] [Q&A] [answered]
Hi! I wanted to bring up an issue where my velocity vectors point in the opposite direction to biological expectation. I know several other similar issues have been raised, and I've tried to use those as a starting point in troubleshooting, but I'm not quite sure how to proceed.

In my case, I am studying maturation of one specific predefined celltype with sampling of timepoints between e14 and p84. We've done some analysis in Monocle 3, and now I'm looking to get a bit more insight into the timing of dynamics. As I'm working with only one celltype with a well-defined linear trajectory, it's pretty straightforward to verify the true direction of the process. I'm working with ~1600 cells. I obtain similar results using the deterministic, stochastic, and dynamic models (dynamic model results shown below). I also obtain similar results even when changing some of the preprocessing parameters (e.g. using n_pcs, neighbours = 10, and subsetting genes either as top 2000 most variable as in the vignette or using the Monocle3-determined differential gene list).

One point to note is that we know that our process involves significant downregulation of genes; about 80% of the differentially expressed genes identified in our pseudotime analysis are downregulated. I'm not sure how relevant this is, but thought I'd share. I'm not fully sure how to interpret the phase plots; and handful of them appear to have the "linear" not curved shape associated with high degradation rates (as discussed on Github) but it doesn't appear to be as many as with other users. I'm also not sure if this is simply a timescale issue, since this process unfolds over days to weeks rather than hours to days as with early differentiation. 

Do you have any recommendations on how I can interpret this data, or other aspects I should look at? Thanks again for all the help!

Velocity plot:
![image](https://user-images.githubusercontent.com/28656813/103553332-50e41180-4e7b-11eb-9793-c8e59671d354.png)

Splicing proportions (this changes a bit depending on which genes I use, but is always going down):
![image](https://user-images.githubusercontent.com/28656813/103553377-5b061000-4e7b-11eb-9857-afa3ac149966.png)

Phase plots for top 20 genes:
![image](https://user-images.githubusercontent.com/28656813/103553424-6ce7b300-4e7b-11eb-9180-364c013c81cf.png)

**Comment by WeilerP** [comment, marked as answer]
The problem is that your phase portraits are non-informative as they do not exhibit the characteristic almond/football shape. Consequently, the model parameters cannot be inferred. Take Atp5g1 for example: The phase portrait is almost linear and a up- and down-regulation has been fitted. We could easily fit only a down-regulation without significantly increasing the loss.

**Comment by pabloatria18** [comment]
@WeilerP I am having similar issues as @skannan4
In my case, I have two different conditions. I tried to run the velocity analysis in both my integrated datasets (using SCTtransform and the Seurat package, first looking just at the control data) and also looking just at the control dataset without integrating. In both cases, I get the arrows pointing from the differentiated clusters towards the progenitor cells.
Is there any recommendation to see if I can overcome this issue?

Thank you very much

**Reply by WeilerP** [reply]
@pabloatria18, in #487 you mention that you expect a differentiation from clusters 0, 1, 3 to cluster 6 (pink) via cluster 2 (green), correct? I am showing your UMAP embedding here again.

<img width="458" alt="UMAP" src="https://user-images.githubusercontent.com/54920011/120822453-6cd00a00-c524-11eb-995f-621bf9af4406.png">

Here are the phase portraits you showed

<img width="659" alt="Top10" src="https://user-images.githubusercontent.com/54920011/120822531-7fe2da00-c524-11eb-884d-737288882357.png">

Couple of thoughts/suggestions/observations:

* Clusters 4 and 5 are not connected to the others. Might be worth excluding them in your analysis. They seem to be quite prominent in your top likelihood genes.
* The phase portrait for *Col13a1* seems to have an almond shape but the temporal ordering or the clusters is off based on your expected differentiation. Based on the phase portrait, cells of cluster 6 are upregulated and partially already in steady state. Following, they are down regulated to cells of cluster 2 during repression phase. Similar reasoning applies to *Fkbp11*.
* Cells from clusters 0, 1, 3 are not really visible in the phase portraits. They seem to be (if at all) not highly abundant in the genes you are showing.

**Reply by pabloatria18** [reply]
Thank you very much!
Yes it is correct. All of them have this transition between the most primitive clusters, the intermediate one and the most differentiated one
I re-run after taking out clusters 4 and 5.
**And I got this:**

<img width="561" alt="Screen Shot 2021-06-07 at 6 52 46 PM" src="https://user-images.githubusercontent.com/54920011/121097174-8fcb1a00-c7c1-11eb-92e9-42bad19afc79.png">

<img width="699" alt="Screen Shot 2021-06-07 at 6 52 13 PM" src="https://user-images.githubusercontent.com/54920011/121097127-7c1fb380-c7c1-11eb-8ab8-e3dd1af04a77.png">

Stochastic, dynamic and steady state look similar
<img width="458" alt="Screen Shot 2021-06-07 at 6 53 10 PM" src="https://user-images.githubusercontent.com/54920011/121097214-9ce80900-c7c1-11eb-9c73-00d3b687707b.png">

<img width="814" alt="Screen Shot 2021-06-07 at 6 58 06 PM" src="https://user-images.githubusercontent.com/54920011/121097638-4d560d00-c7c2-11eb-89b2-d1be8af1cde6.png">


**Then I removed Cluster 3 due to the spliced / unspliced ratio and I got this:**

<img width="711" alt="Screen Shot 2021-06-07 at 6 55 31 PM" src="https://user-images.githubusercontent.com/54920011/121097415-f0f2ed80-c7c1-11eb-89a6-24678b6cc67b.png">

<img width="465" alt="Screen Shot 2021-06-07 at 6 55 46 PM" src="https://user-images.githubusercontent.com/54920011/121097438-f9e3bf00-c7c1-11eb-9d62-67a4d98c4714.png">

<img width="799" alt="Screen Shot 2021-06-07 at 7 02 40 PM" src="https://user-images.githubusercontent.com/54920011/121097994-f270e580-c7c2-11eb-9238-96d95d4698e8.png">


For both of them I am using:
min_shared_counts=10
n_top_genes=2000
n_pcs=30
n_neighbors=30

Do you have any advice or ideas?

Thank you once again!

**Reply by WeilerP** [reply]
Sorry, I meant only taking out clusters `4` and `5` but **not** recalculate the UMAP embedding and clustering.
I also just realized that cluster `4` seems to be split into two parts - one isolated and one connected to the big connected component. Might be that these observations are different to one another and should be assigned individual clusters?
Also, please retain the original color map from your first run. This will make the discussion much easier since the color-cluster mapping is unique.

**Reply by pabloatria18** [reply]
@WeilerP Thank you!
So I re run it like this. I am not sure if I am doing the subset right (I mean after running the scv.pp.filter_and_normalize and scv.pp.moments)

First I ran: 

scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000) #min_shared_counts=20,
scv.pp.moments(adata, n_pcs=30, n_neighbors=30)

Then I subset the clusters like this:
adata = adata[adata.obs['Clusters'].isin(['0', '1', '2', '3', '6', '7'])].copy()

Then I ran the velocity:
scv.tl.velocity(adata, mode='stochastic')
scv.tl.velocity_graph(adata)
<img width="254" alt="Subset 1" src="https://user-images.githubusercontent.com/54920011/121577313-64317500-c9f7-11eb-95f5-9d0e47aad5c3.png">

scv.tl.velocity(adata, mode='steady_state')
<img width="253" alt="Subset2" src="https://user-images.githubusercontent.com/54920011/121577404-7ad7cc00-c9f7-11eb-96b9-b493ee701a60.png">

scv.tl.velocity(adata, mode='dynamical')
<img width="253" alt="Subset3" src="https://user-images.githubusercontent.com/54920011/121577489-8dea9c00-c9f7-11eb-80c4-5e095b8b5961.png">

top_genes = adata.var.sort_values('fit_likelihood', ascending=False).index[:10]
scv.pl.scatter(adata, basis=top_genes, color="Clusters", wspace=1, hspace=1, nrows=2, ncols=5)
<img width="664" alt="Top10subset" src="https://user-images.githubusercontent.com/54920011/121577580-a78be380-c9f7-11eb-9928-0a2a70908cc1.png">

Thank you!

**Reply by WeilerP** [reply]
LGTM (code wise)! My comments from above (partially) still apply, though:

* Cells from clusters 0, 1, 3 are not really visible in the phase portraits. They seem to be (if at all) not highly abundant in the genes you are showing. Instead, clusters 6 (pink) and 2 (green) are quite spread out in the phase portraits.
* There really isn't much curvature in most of the phase portraits you are showing. Linear phase portraits can, for example, be interpreted as up, down or up and down regulations.
* It looks like a down regulation was fitted for *Fkbp11* which violates the temporal ordering know from biology. *Cpe* and *Car* look similar (starting from cluster 6 (pink) in steady state then differentiating into cluster 2 (green) during downregulation. *Cgref1* should be discarded all together, IMO, since cells from cluster 6 (pink) distort everything with being almost disconnected from the rest.

Seems to me like RNA velocity cannot be used on your dataset.

**Reply by pabloatria18** [reply]
@WeilerP I appreciate your help!
That's unfortunate.
I will give it a try with other available methods!

**Reply by WeilerP** [reply]
@pabloatria18, you could try out [CellRank](https://cellrank.readthedocs.io/en/stable/) with the CytoTRACE kernel. On `cellrank@develop`, you'll also already find a Waddington OT kernel.

**Reply by pabloatria18** [reply]
@WeilerP I did the CytoTRACE kernel and looks better! 
Due to my inexperience, I am wondering why some datasets work and some others do not with the RNA velocity. Do you think this has biological relevance? I mean, I don't want to bias my interpretation of the data just because the arrows are not pointing where I am expecting them to...

**Reply by denvercal1234GitHub** [reply]
@pabloatria18 -- Did you figure out how to do the CytoTRACE kernel? Is there any things you needed to modify from the tutorials (https://cellrank.readthedocs.io/en/stable/)?

**Comment by BW15061999** [comment]
Hi @WeilerP,
This is the top10 phase portraits of my results, but the direction of the results is from 7 day to 4.5 day. Although most of the 7d cells are memory cells, which are quite possible for them to differentiate back to 4.5d effector cells in real stiuation.
I am not quite sure if phase portraits are informative enough to run RNA velocity because some of them, in my view, exhibit almond/football shape, but some of the phase portraits are quite linear.
It would be great if you can have a look at my phase portraits!  Thank you!

Best
![download](https://user-images.githubusercontent.com/52130151/182756870-c24bdc1b-864d-42b3-8901-506210149cd5.png)
![Screenshot from 2022-08-04 11-28-05](https://user-images.githubusercontent.com/52130151/182839960-776cda4d-5de5-4e4c-9be9-da17d61a3d65.png)
![Screenshot from 2022-08-04 11-28-20](https://user-images.githubusercontent.com/52130151/182839982-2d254372-29db-43d8-a398-452efcbd3efe.png)

**Comment by DaianeH** [comment]
Hi @WeilerP,

I had a problem when running scVelo on my datasets and different results were observed between stochastic and dynamic models. I'm looking into the differentiation of progenitors into myeloid cells. In blue are progenitors, in orange GMPs, and in green monocytes. These are the top10  phase portraits of my results, and indeed I don't see the almond/football shape in them.

[scvelo_top_genes_LEED-D5-dynamic.pdf](https://github.com/theislab/scvelo/files/10252082/scvelo_top_genes_LEED-D5-dynamic.pdf)

How exactly can I interpret results when there is no almond/football shape? I ask because I see little difference from the results in your example dataset on scvelo tutorial: https://scvelo.readthedocs.io/en/stable/DynamicalModeling/#Top-likelihood-genes

In addition, can I infer from these results that my dataset is not suitable for RNA velocity, and if so why? Because of low data quality?

**Reply by WeilerP** [reply]
@DaianeH, it looks like the phase portraits do not show a transition between the different cell types. For RNA velocity to work, you are expecting to see the different cell types ordered along the phase portrait (e.g. blue to orange to green). Based on the phase portraits you shared, I'd say the data is not suitable for RNA velocity analysis.

**Reply by denvercal1234GitHub** [reply]
Hi @WeilerP Did you mean that in the phase portraits above, for each of the genes shown, the cells are simply clumped together along the diagonal line (suggestive of a linear relationship) instead of the characteristic almond shape (where the cells from blue clusters, followed by the orange cluster cells then the green cluster cells curve up above the diagonal line, and have some cluster cells curved down beneath the line for down regulation)? 

Though, wouldn't the gene `Gm26740` above shows this characteristic shape?

Thank you for your help.

**Comment by flde** [comment]
Dear all, I often observe that velocity is reversed in biological systems including progenitor states. The CytoTRACE framework is based on the observation that progenitor cells express more features compared to mature cells. Hence, during early differentiation it might be that more genes are silenced compared to activation. Would that be a possible biological explanation for the reversed velocity stream?
